### DISCUSSION TITLE
[Proposal] 2 game mode ideas. Are you considering any?

### DISCUSSION THREAD

**Discussion by Epicm14** [ppy/osu#12991] [Ideas]
Sorry if you're not accepting any, I never read anywhere that you weren't, but I have 2 idea that I like.
Also not sure if this is the right place to put this. Ideas give birth to better ideas, right?
One is complex, the other is simple

**1st one** (simple one):

**osu!lazer** (yes it's the name of this mode idea)
This idea sounds like a very fitting and awesome way to release the finished osu!lazer you're working on.

idea: https://prntscr.com/kf7af3

- A lazer themed mode where you have a lazer shooter in the middle of the screen(a hitcircle), shooting lazers, doesn't move. Or it can be a lazer somehow.
- Perimeter is your screen, and it is shrinking with HP.
- Portions of the perimeter light up at certain times, and you have to aim for them.
- Aim your shooter using cursor.
- Rhythm is closing circles around shooter, just like standard.(Feels better to have timing bars going out to the perim. and lighting up when hit.
- Hit with zx or somthing.
- When perimeter shrinks to a certain size, you fail.
- Failing to hit the lighted perimeter is a miss.
- Lazers move fast, so lights can disappear quickly.
- Also considered two different color lights showing on the perimeter(ex. red and blue), one light using z and x, the other using a and s, or something. Just a way to make it more complex.

**2nd one:**
(as a second thought, this one sounds too unrealistic, but better ideas can come from it)
**osu!shoot** (complex one)

tl;dr Basically a bullet hell shooter with rhythm

idea: https://prntscr.com/kf6vkg

- Single player has cpu's with varying numbers, size and skill with * level
- Multiplayer uses other players, can be free for all, or teams. 
- Is it possible for maps to be extended past 1 screen (to accommodate for 16 players)?
- You have a beat that you shoot bullets to. Closing rings around cursor is the beat, like standard is with hitcircles. (I also considered a line to the side like 1k mania)
- CPU also shoots to the beat.
- Can gain extra shots when players/CPUs are hit.(?)
- Cursors are all the same size in multi for fairness 
- When hit you lose hp, drop combo, miss notes, cursor runs into another cursor( You get points for running into a player/CPU because that's a hit, but the losses make it not worth it.)
- When HP is drained completely,  die(for multi), fail(for single)
- I'd say score should be something like x1.05 when you hit other players/CPU with bullets. It should give a bit because of difficulty.
- CPUs can have HP drained and die, you will get more points for that. (I don't think they should be 
  replaced, but they should have a reasonable HP for the amount of notes.)
- No gaining hp because that would be too easy
- Combo for hitting notes(the harder it is, the more points combo gives)
- CPU cursors have points where they stop moving?
- Minimun value of other cursors hit for score to count?
- Break combo when being hit, missing notes 
- Lose acc when hit, mistime or miss notes 
- 300 100 50 X   acc   sounds best
- Combo scoring sounds best imo, don't see how acc scoring would work
          -rely on hitting other cursors for a big score 
          -additional small score for enemy bullets that fly off the screen
- Have a dotted line showing where you aim at all times.(like an aim marker)
- Aim the line using two buttons, one for clockwise, the other counterclockwise.(ex. if you use z and x for 
  hitting, use left Shift and Space for aiming, or whatever fits.)
- Aim with any controller, but no teleporting, so touch would be awkward.
- Mods that give/take off score include: 
             -EZ: bigger and slower CPUs, smaller user cursor, slower bullets.
             -A mode-only mod with walls for protection.
             -HR: Faster bullets, bugger user cursor, smaller and faster CPU cursors change size of cursors, 
                     bullets, have hidden bullets.
             -HD: 2 versions; hidden CPUs, small increase; hidden bullets(but are hinted at when shot), large 
                     increase
- Your cursor never leaves the screen 
- Long notes are a thing. (Making some super long notes that divide the game, and can be aimed like noodles sound cool.)
- Spinners would cease gameplay for a short period of time but are a thing  

I'm not a mapper so I'm not sure how mappable this would be.

**Comment by Tanza3D** [comment]
It would be somewhat hard to go around the place (website code, game code, ect) adding these new images for these new gamemodes and also adding the online scoring code, but it would be possible and i really like these ideas,
